src/test/java/com/google/spanner/liquibase/TestHarness.java

```diff
@@ -0,0 +1,259 @@
+                    .withStartupTimeout(Duration.ofSeconds(10))
+                    .waitingFor(Wait.forHttp("/").forStatusCode(404));
+
+            // Start the container
+            testContainer.start();
+
+            // JDBC Connection
+            spannerEmulatorHost = String.format("%s:%d",
+                    testContainer.getContainerIpAddress(),
+                    testContainer.getMappedPort(9010));
+
+        }
+
+        // Create the Spanner service
+        final Spanner service = SpannerOptions.newBuilder()
+                .setProjectId(PROJECT_ID)
+                .setEmulatorHost(String.format("http://%s", spannerEmulatorHost))
+                .build()
+                .getService();
+
+        // Initialize the instance and database.
+        createInstance(service, INSTANCE_ID);
+        createDatabase(service, INSTANCE_ID, DATABASE_ID);
+
+        // JDBC Connection
+        final java.sql.Connection conn = DriverManager.getConnection(
+                String.format(
+                        "jdbc:cloudspanner://%s/projects/%s/instances/%s/databases/%s?autocommit=false;usePlainText=true",
+                        spannerEmulatorHost, PROJECT_ID, INSTANCE_ID, DATABASE_ID));
+
+        return new Connection() {
+
+            @Override
+            public java.sql.Connection getJDBCConnection() {
+                return conn;
+            }
+
+            @Override
+            public void stop() { }
+        };
+    }
+
+    //
+    // Create a normal Spanner connection.
+    //
+    // Automatically creates and drops a temporary liquibase database
+    //
+    public static Connection useSpannerConnection() throws SQLException {
+        final String projectId = System.getenv("SPANNER_PROJECT");
+        final String instanceId = System.getenv("SPANNER_INSTANCE");
+        final String databaseId = String.format("test_database_%d", new Random().nextInt(1_000_000));
+
+        // Ensure parameters are properly set
+        if (projectId == null || instanceId == null) {
+            throw new SQLException("Both SPANNER_PROJECT and SPANNER_INSTANCE environment variables must be set!");
+        }
+
+        Spanner service = SpannerOptions.newBuilder()
+                .setProjectId(projectId)
+                .build()
+                .getService();
+
+        createDatabase(service, instanceId, databaseId);
+
+        // JDBC connection initialize
+        java.sql.Connection conn = DriverManager.getConnection(
+                String.format(
+                    "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s?autocommit=true",
+                    projectId, instanceId, databaseId));
+
+        return new Connection() {
+
+            @Override
+            public java.sql.Connection getJDBCConnection() {
+                return conn;
+            }
+
+            @Override
+            public void stop() throws SQLException {
+                dropDatabase(service, instanceId, databaseId);
```

Comment:
May wish to check for specific exceptions when dropping a database and retry if needed

Reply:
As it is a test harness and any exceptions will cause the test to fail, I'm less concerned about. This is the cleanup of a database.